### DISCUSSION TITLE
Trying to understand the system design of Blockscout

### DISCUSSION THREAD

**Discussion by jhkimqd** [blockscout/blockscout#7252] [Q&A]
### On a high-level,
1. Blockscout indexes blockchain data
2. The data is saved to a relational database
3. The data from RDS is queried and used to display the explorer dashboard

### Question:
1. I can't seem to find _when_ (at what point, e.g. after each transaction is made) Blockscout indexes the data and saves it onto the RDS. 
2. Among many archive nodes, which particular one does Blockscout 'attach' and crawl the block data?

**Comment by vbaranov** [comment]
Yes, on a high level, your understanding is correct.

> 1. I can't seem to find when (at what point, e.g. after each transaction is made) Blockscout indexes the data and saves it onto the RDS.

Blockscout contains 2 main block fetchers:
- realtime block fetcher, which subscribes to new block heads through websocket endpoint of the archive node or by polling `eth_getBlockNumber` method of nodes's JSON RPC endpoint and then fetch all possible block's data to the DB.
- catchup block fetcher, which constantly works from the head of the chain to the tail, by fetching blocks in batches from the node, which were not indexed yet in Blockscout DB.

> 2. Among many archive nodes, which particular one does Blockscout 'attach' and crawl the block data?

The list of supported by Blockscout EVM execution layer clients is here https://docs.blockscout.com/for-developers/information-and-settings/client-settings.
